ocsmesh/mesh/mesh.py

```diff
@@ -662,13 +662,23 @@ def interpolate(self, raster: Union[Raster, List[Raster]],
         nprocs = -1 if nprocs is None else nprocs
         nprocs = cpu_count() if nprocs == -1 else nprocs
 
-        with Pool(processes=nprocs) as pool:
-            res = pool.starmap(
-                _mesh_interpolate_worker,
-                [(self.vert2['coord'], self.crs,
-                    _raster.tmpfile, _raster.chunk_size, method)
-                 for _raster in raster]
-                )
+        # Fix an issue on Jupyter notebook where having pool execute
+        # interpolation even in case of nprocs == 1 would results in
+        # application getting stuck
+        if nprocs > 1:
+            with Pool(processes=nprocs) as pool:
+                res = pool.starmap(
+                    _mesh_interpolate_worker,
+                    [(self.vert2['coord'], self.crs,
+                        _raster.tmpfile, _raster.chunk_size, method)
+                     for _raster in raster]
+                    )
+            pool.join()
+        else:
+            res = [_mesh_interpolate_worker(
+                        self.vert2['coord'], self.crs,
+                        _raster.tmpfile, _raster.chunk_size, method)
+                   for _raster in raster]
```

Comment:
Using pool of processes can cause issues in Jupyter lab, even when 1 process is used. The issue leads to the application getting stuck during interpolation. To avoid the issue if the # procs is 1, don't use pool.